From 14ff73f07783fbbb9c40b77add75e68bc5f10ec6 Mon Sep 17 00:00:00 2001
From: Matteo Beccati <matteo@beccati.com>
Date: Tue, 27 Sep 2016 11:22:18 +0200
Subject: [PATCH] Fix h1 report 170156

Reflected XSS
-------------

The HackerOne user pavanw3b has reported that the Revive Adserver web
installer scripts were vulnerable to a reflected XSS attack via the dbHost,
dbUser and possibly other parameters. It has to be noted that the window
for such attack vectors to be possible is extremely narrow and it is very
unlikely that such an attack could be actually effective.

CWE: CWE-79
CVSSv2: 4 (AV:N/AC:H/Au:N/C:P/I:P/A:N)

CVSSv3 Vector: CVSS:3.0/AV:N/AC:H/PR:N/UI:R/S:U/C:L/I:L/A:N/E:U/RL:O/RC:C
CVSSv3 Base Score: 4.2
CVSSv3 Temporal Score: 3.7
---
 lib/templates/wizard/install/messages.html | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/templates/wizard/install/messages.html b/lib/templates/wizard/install/messages.html
index bf099539dc..9b0dbf8395 100644
--- a/lib/templates/wizard/install/messages.html
+++ b/lib/templates/wizard/install/messages.html
@@ -6,7 +6,7 @@
           <div class="body">
 			<span id='errorMessages'>
             {foreach from=$aMessages.error item=error name=messagesLoop}
-              {$error} {if !$smarty.foreach.messagesLoop.last}<br/>{/if}
+              {$error|escape} {if !$smarty.foreach.messagesLoop.last}<br/>{/if}
             {/foreach}
             </span>
           </div>
